lib/spack/spack/test/concretize_preferences.py

```diff
@@ -84,6 +85,16 @@ def test_preferred_variants(self):
             'mpileaks', debug=True, opt=True, shared=False, static=False
         )
 
+    def test_preferred_variants_from_any(self):
+        update_packages('multivalue-variant', 'variants', 'foo=bar')
+        assert_variant_values(
+            'multivalue-variant foo=any', foo=('bar',)
```

Comment:
Could this be considered fragile since technically any choice of foo would be correct? Or is it because 'bar' is mentioned first in the variant values?

Reply:
It's because the previous line sets the default to the single value `bar`.